servers/rendering/renderer_rd/storage_rd/render_scene_buffers_rd.h

```diff
@@ -0,0 +1,273 @@
+			context = p_context;
+			buffer_name = p_texture_name;
+		}
+	};
+
+	struct NamedTexture {
+		// Cache the data used to create our texture
+		RD::TextureFormat format;
+		bool is_unique; // If marked as unique, we return it into our pool
+
+		// Our texture objects, slices are lazy (i.e. only created when requested).
+		RID texture;
+		Vector<RID> slices;
+	};
+
+	mutable HashMap<NTKey, NamedTexture, NTKey> named_textures;
+	void free_named_texture(NamedTexture &p_named_texture);
+
+	// Data buffers
+	mutable HashMap<StringName, Ref<RenderSceneBufferDataRD>> data_buffers;
+
+protected:
+	static void _bind_methods();
+
+public:
+	RenderSceneBuffersRD();
+	virtual ~RenderSceneBuffersRD();
+
+	// info from our renderer
+	void set_can_be_storage(const bool p_can_be_storage) { can_be_storage = p_can_be_storage; }
+	void set_max_cluster_elements(const uint32_t p_max_elements) { max_cluster_elements = p_max_elements; }
+	void set_base_data_format(const RD::DataFormat p_base_data_format) { base_data_format = p_base_data_format; }
+	RD::DataFormat get_base_data_format() const { return base_data_format; }
+	void set_sseffects(RendererRD::SSEffects *p_ss_effects) { sse = p_ss_effects; }
+	void set_vrs(RendererRD::VRS *p_vrs) { vrs = p_vrs; }
+
+	void cleanup();
+	virtual void configure(RID p_render_target, const Size2i p_internal_size, const Size2i p_target_size, float p_fsr_sharpness, float p_texture_mipmap_bias, RS::ViewportMSAA p_msaa, RenderingServer::ViewportScreenSpaceAA p_screen_space_aa, bool p_use_taa, bool p_use_debanding, uint32_t p_view_count) override;
+	virtual void set_fsr_sharpness(float p_fsr_sharpness) override;
+	virtual void set_texture_mipmap_bias(float p_texture_mipmap_bias) override;
+	virtual void set_use_debanding(bool p_use_debanding) override;
+
+	// Named Textures
+
+	bool has_texture(const StringName &p_context, const StringName &p_texture_name) const;
+	RID create_texture(const StringName &p_context, const StringName &p_texture_name, const RD::DataFormat p_data_format, const uint32_t p_usage_bits, const RD::TextureSamples p_texture_samples = RD::TEXTURE_SAMPLES_1, const Size2i p_size = Size2i(0, 0), const uint32_t p_depth = 1, const uint32_t p_layers = 0, const uint32_t p_mipmaps = 1, RD::TextureView p_view = RD::TextureView(), bool p_unique = true);
+	RID create_texture_view(const StringName &p_context, const StringName &p_texture_name, const StringName p_view_name, RD::TextureView p_view = RD::TextureView());
+	RID get_texture(const StringName &p_context, const StringName &p_texture_name) const;
+	const RD::TextureFormat get_texture_format(const StringName &p_context, const StringName &p_texture_name) const;
+	RID get_texture_slice(const StringName &p_context, const StringName &p_texture_name, const uint32_t p_layer, const uint32_t p_mipmap);
+
+	void clear_context(const StringName &p_context);
+
+	// Data buffers
+	bool has_data_buffer(const StringName &p_name);
```

Comment:
I think I would rename this to custom_data / RenderBuffersCustomData, because in general this can contain more things than just buffers.

Reply:
More than happy to apply the new names:)